src/pint/scripts/event_optimize.py

```diff
@@ -758,6 +764,18 @@ def main(argv=None):
 
     import emcee
 
+    # Setting up a backend to save the chains into an h5 file
+    try:
+        if backendpath:
+            backend = emcee.backends.HDFBackend(
```

Comment:
It's preferable to construct the filename in the `if` statement so as not to duplicate the HDF5 construction code.